v2/cmd/snapshot/snapshot.go

```diff
@@ -0,0 +1,79 @@
+package snapshot
+
+import (
+	"os"
+	"time"
+
+	"github.com/cosmos/iavl/v2"
+	"github.com/rs/zerolog"
+	zlog "github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+)
+
+var log = zlog.Output(zerolog.ConsoleWriter{
+	Out:        os.Stderr,
+	TimeFormat: time.Stamp,
+})
+
+func Command() *cobra.Command {
+	var (
+		version int64
+		dbPath  string
+	)
+	cmd := &cobra.Command{
+		Use:   "snapshot",
+		Short: "take a snapshot of the tree at version n and write to SQLite",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			paths, err := iavl.FindDbsInPath(dbPath)
+			if err != nil {
+				return err
+			}
+			log.Info().Msgf("found db paths: %v", paths)
+
+			var (
+				pool   = iavl.NewNodePool()
+				done   = make(chan struct{})
+				errors = make(chan error)
+				cnt    = 0
+			)
+			for _, path := range paths {
+				cnt++
+				sqlOpts := iavl.SqliteDbOptions{Path: path}
+				sql, err := iavl.NewSqliteDb(pool, sqlOpts)
```

Comment:
This handle is never closed on unrelated errors. Please close it after use perhaps with a defer, or in each error flow clause.